src/test/java/com/neo4j/docker/neo4jserver/TestHealthcheck.java

```diff
@@ -0,0 +1,112 @@
+package com.neo4j.docker.neo4jserver;
+
+import com.neo4j.docker.neo4jserver.configurations.Configuration;
+import com.neo4j.docker.neo4jserver.configurations.Setting;
+import com.neo4j.docker.utils.TemporaryFolderManager;
+import com.neo4j.docker.utils.TestSettings;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+public class TestHealthcheck
+{
+    private static Logger log = LoggerFactory.getLogger( TestHealthcheck.class );
+    private static Map<Setting,Configuration> confNames = Configuration.getConfigurationNameMap();
+
+    @RegisterExtension
+    public static TemporaryFolderManager temporaryFolderManager = new TemporaryFolderManager();
+
+    private GenericContainer createContainerWithDefaultListenAddress()
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+        return container;
+    }
+
+    private GenericContainer createContainerWithEnvVarListenAddress( String listenAddress )
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withEnv( confNames.get( Setting.HTTP_LISTEN_ADDRESS ).envName, listenAddress )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+        return container;
+    }
+
+    private GenericContainer createContainerWithConfigListenAddress( String listenAddress ) throws IOException
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+
+        var tempConfigDir = temporaryFolderManager.createTempFolder( "temp_neo4j_config" );
```

Comment:
Running these docker tests tends to create a LOT of folders in `local-mounts`, so it's helpful for the temporary folders to be named in a way that means we can trace the temporary folder back to the test that generated it.

I don't have a good system for naming test folders, and there's DEFINITELY room for improvement in how the test folders are structured and named.
`"temp_neo4j_config"` is just too vague a name though. The `temp_` prefix is redundant, and it should at least mention the word `healthcheck`.